core/src/main/java/org/apache/calcite/sql/type/ReturnTypes.java

```diff
@@ -427,6 +427,13 @@ public static SqlCall stripSeparator(SqlCall call) {
   public static final SqlReturnTypeInference VARCHAR_2000_NULLABLE =
       VARCHAR_2000.andThen(SqlTypeTransforms.TO_NULLABLE);
 
+  /**
+   * Type-inference strategy that always returns "VARCHAR" with nulls
+   * allowed if any of the operands allow nulls.
+   */
+  public static final SqlReturnTypeInference VARCHAR_NULLABLE =
```

Comment:
Can we declare another `VARCHAR`?

Because
`VARCHAR_4` -> `VARCHAR_4_NULLABLE`;
`VARCHAR_2000` -> `VARCHAR_2000_NULLABLE`;

Reply:
Thanks for reviewing, updated